R/data.table.R

```diff
@@ -1029,8 +1029,14 @@ replace_dot_alias = function(e) {
           if (is.name(lhs)) {
             lhs = as.character(lhs)
           } else {
-            # e.g. (MyVar):= or get("MyVar"):=
-            lhs = eval(lhs, parent.frame(), parent.frame())
+            #i.e lhs is names(.SD) || setdiff(names(.SD), cols) || (cols)
+            replace_names_sd = function(e, cols){
+              if (length(e) == 1L) return(e)
+              if (e[[1L]] == quote(names) && e[[2L]] == quote(.SD)) return(cols)
```

Comment:
I think we need an `is.name(e[[2L]])` or this would apply to `names(.SD(4))`?

Reply:
Added although may not be necessary. Regardless, should be more robust.
```
quote(names(.SD(4)))[[2L]]
## .SD(4)
quote(names(.SD(4)))[[2L]] == quote(.SD)
## [1] FALSE
```